[improve]: Log Issue Only Once

### Improvement target

Logging Statements
Example of duplicated statements:
````
\",\"log\":{\"level\":\"ERROR\",\"logger\":\"com.bytechef.tenant.TenantContext\"},\"process\":{\"pid\":1,\"thread\":{\"name\":\"tomcat-handler-50713\"}},\"service\":{\"name\":\"server-app\",\"version\":\"0.26.0\",\"node\":{}},\"message\":\"Unable to execute call with tenant\",\"traceId\":\"dc6082efa6f86dae4b3fa0baa4a180b8\",\"spanId\":\"7641b2e0af970ef9\",\"tenantId\":\"\",\"ecs\":{\"version\":\"8.11\"}}\n","stream":"stdout","time":"2026-05-05T20:26:26.662764332Z"}
{"log":"{\"@timestamp\":\"2026-05-05T20:26:26.664248171Z\",\"log\":{\"level\":\"ERROR\",\"logger\":\"com.bytechef.web.rest.error.GlobalResponseEntityExceptionHandler\"},\"process\":{\"pid\":1,\"thread\":{\"name\":\"tomcat-handler-50713\"}},\"service\":{\"name\":\"server-app\",\"version\":\"0.26.0\",\"node\":{}},\"message\":\"java.lang.IllegalArgumentException: Approval already processed\",\"traceId\":\"dc6082efa6f86dae4b3fa0baa4a180b8\",\"spanId\":\"7641b2e0af970ef9\",\"tenantId\":\"public\",\"error\":{\"type\":\"java.lang.RuntimeException\",\"message\":\"java.lang.IllegalArgumentException: Approval already processed\",\"stack_trace\":\"java.lang.RuntimeException: java.lang.IllegalArgumentException: Approval already processed\\n\\tat com.bytechef.tenant.TenantContext.callWithTenantId(TenantContext.java:48)\\n\\tat com.bytechef.platform.webhook.web.rest.ApprovalController.approve(ApprovalController.java:60)\\n\\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\\n\\tat java.base/java.lang.reflect.Method.invoke(Method.java:565)\\n\\tat org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:252)\\n\\tat org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:184)\\n\\tat org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)\\n\\tat org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:934)\\n\\tat org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:853)\\n\\tat org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:86)\\n\\tat org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:963)\\n\\tat org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:866)\\n\\tat org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1000)\\n\\tat org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:892)\\n\\tat jakarta.servlet.http.HttpServlet.service(HttpServlet.java:633)\\n\\tat org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:874)\\n\\tat jakarta.servlet.http.HttpServlet.service(HttpServlet.java:723)\\n\\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:128)\\n\\tat org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)\\n\\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:107)\\n\\tat com.bytechef.ee.observability.filter.AddTraceIdFilter.doFilterInternal(AddTraceIdFilter.java:50)\\n\\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)\\n\\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:107)\\n\\tat com.bytechef.ee.observability.filter.HeaderLoggerFilter.doFilterInternal(HeaderLoggerFilter.java:43)\\n\\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)\\n\\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:107)\\n\\tat org.springframework.web.servlet.resource.ResourceUrlEncodingFilter.doFilter(ResourceUrlEncodingFilter.java:66)\\n\\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:107)\\n\\tat org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:108)\\n\\tat org.springframework.security.web.FilterChainProxy.lambda$doFilterInternal$3(FilterChainProxy.java:235)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$FilterObservation$SimpleFilterObservation.lambda$wrap$1(ObservationFilterChainDecorator.java:493)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$AroundFilterObservation$SimpleAroundFilterObservation.lambda$wrap$1(ObservationFilterChainDecorator.java:354)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator.lambda$wrapSecured$0(ObservationFilterChainDecorator.java:86)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:132)\\n\\tat org.springframework.security.web.access.intercept.AuthorizationFilter.doFilter(AuthorizationFilter.java:101)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:231)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:126)\\n\\tat org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:120)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:231)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:100)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:231)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:181)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:231)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:231)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat com.bytechef.security.web.filter.SpaWebFilter.doFilterInternal(SpaWebFilter.java:88)\\n\\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:231)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:110)\\n\\tat org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:96)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:231)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:91)\\n\\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:231)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90)\\n\\tat org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75)\\n\\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:231)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat org.springframework.security.web.context.SecurityContextHolderFilter.doFilter(SecurityContextHolderFilter.java:82)\\n\\tat org.springframework.security.web.context.SecurityContextHolderFilter.doFilter(SecurityContextHolderFilter.java:69)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:231)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:62)\\n\\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:231)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat org.springframework.security.web.session.DisableEncodeUrlFilter.doFilterInternal(DisableEncodeUrlFilter.java:42)\\n\\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:244)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$AroundFilterObservation$SimpleAroundFilterObservation.lambda$wrap$0(ObservationFilterChainDecorator.java:337)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:228)\\n\\tat org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:141)\\n\\tat org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:237)\\n\\tat org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:195)\\n\\tat org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:113)\\n\\tat org.springframework.web.filter.ServletRequestPathFilter.doFilter(ServletRequestPathFilter.java:52)\\n\\tat org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:113)\\n\\tat org.springframework.web.filter.CompositeFilter.doFilter(CompositeFilter.java:74)\\n\\tat org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration$CompositeFilterChainProxy.doFilter(WebSecurityConfiguration.java:317)\\n\\tat org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:355)\\n\\tat org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:272)\\n\\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:107)\\n\\tat org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)\\n\\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)\\n\\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:107)\\n\\tat org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)\\n\\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)\\n\\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:107)\\n\\tat org.springframework.web.filter.ServerHttpObservationFilter.doFilterInternal(ServerHttpObservationFilter.java:110)\\n\\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)\\n\\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:107)\\n\\tat org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:199)\\n\\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)\\n\\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:107)\\n\\tat org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:165)\\n\\tat org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:77)\\n\\tat org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:492)\\n\\tat org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:113)\\n\\tat org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:83)\\n\\tat org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:72)\\n\\tat org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:341)\\n\\tat org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)\\n\\tat org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)\\n\\tat org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:903)\\n\\tat org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1801)\\n\\tat org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)\\n\\tat java.base/java.lang.VirtualThread.run(VirtualThread.java:456)\\nCaused by: java.lang.IllegalArgumentException: Approval already processed\\n\\tat com.bytechef.atlas.execution.facade.JobFacadeImpl.resumeApproval(JobFacadeImpl.java:120)\\n\\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\\n\\tat java.base/java.lang.reflect.Method.invoke(Method.java:565)\\n\\tat org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:359)\\n\\tat org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:715)\\n\\tat com.bytechef.atlas.execution.facade.JobFacadeImpl$$SpringCGLIB$$0.resumeApproval(\u003cgenerated\u003e)\\n\\tat com.bytechef.platform.webhook.web.rest.ApprovalController.lambda$approve$0(ApprovalController.java:61)\\n\\tat com.bytechef.tenant.TenantContext.callWithTenantId(TenantContext.java:44)\\n\\t... 126 common frames omitted\\n\"},\"ecs\":{\"version\":\"8.11\"}}\n","stream":"stdout","time":"2026-05-05T20:26:26.666810044Z"}
{"log":"{\"@timestamp\":\"2026-05-05T20:26:26.666927175Z\",\"log\":{\"level\":\"ERROR\",\"logger\":\"com.bytechef.web.rest.error.GlobalResponseEntityExceptionHandler\"},\"process\":{\"pid\":1,\"thread\":{\"name\":\"tomcat-handler-50713\"}},\"service\":{\"name\":\"server-app\",\"version\":\"0.26.0\",\"node\":{}},\"message\":\"Approval already processed\",\"traceId\":\"dc6082efa6f86dae4b3fa0baa4a180b8\",\"spanId\":\"7641b2e0af970ef9\",\"tenantId\":\"public\",\"error\":{\"type\":\"java.lang.IllegalArgumentException\",\"message\":\"Approval already processed\",\"stack_trace\":\"java.lang.IllegalArgumentException: Approval already processed\\n\\tat com.bytechef.atlas.execution.facade.JobFacadeImpl.resumeApproval(JobFacadeImpl.java:120)\\n\\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\\n\\tat java.base/java.lang.reflect.Method.invoke(Method.java:565)\\n\\tat org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:359)\\n\\tat org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:715)\\n\\tat com.bytechef.atlas.execution.facade.JobFacadeImpl$$SpringCGLIB$$0.resumeApproval(\u003cgenerated\u003e)\\n\\tat com.bytechef.platform.webhook.web.rest.ApprovalController.lambda$approve$0(ApprovalController.java:61)\\n\\tat com.bytechef.tenant.TenantContext.callWithTenantId(TenantContext.java:44)\\n\\tat com.bytechef.platform.webhook.web.rest.ApprovalController.approve(ApprovalController.java:60)\\n\\tat java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)\\n\\tat java.base/java.lang.reflect.Method.invoke(Method.java:565)\\n\\tat org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:252)\\n\\tat org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:184)\\n\\tat org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)\\n\\tat org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:934)\\n\\tat org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:853)\\n\\tat 
````

### Improvement description

Avoid logging same issue more than once. If method throws exception leaning on causedBy mechanism then the last exception catch in the chain should log Error message.
If it is necessary to log-out particular detail available in the scope of the method that handles exception than log out details in exception message and add originating exception as causedBy argument of a new exception.

### Observed Version

0.27.0

--- a/server/ee/libs/config/tenant-multi-data-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantDataSource.java
+++ b/server/ee/libs/config/tenant-multi-data-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantDataSource.java
@@ -31,7 +31,7 @@ public class MultiTenantDataSource implements DataSource {
 
     private final DataSource dataSource;
     private static final Logger log = LoggerFactory.getLogger(MultiTenantDataSource.class);
-    private static final String SET_SEARCH_PATH_STATEMENT = "SET search_path TO ?";
+    private static final String SET_SEARCH_PATH_STATEMENT = "SET search_path TO ";
 
     @SuppressFBWarnings("EI")
     public MultiTenantDataSource(DataSource dataSource) {
@@ -109,8 +109,8 @@ public class MultiTenantDataSource implements DataSource {
     private static void setSearchPath(Connection connection) throws SQLException {
         String currentDatabaseSchema = TenantContext.getCurrentDatabaseSchema();
 
-        try (PreparedStatement statement = connection.prepareStatement(SET_SEARCH_PATH_STATEMENT)) {
-            statement.setString(1, currentDatabaseSchema);
+        try (PreparedStatement statement =
+            connection.prepareStatement(SET_SEARCH_PATH_STATEMENT + currentDatabaseSchema)) {
 
             statement.execute();
 

--- a/server/ee/libs/config/tenant-multi-data-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantDataSource.java
+++ b/server/ee/libs/config/tenant-multi-data-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantDataSource.java
@@ -12,11 +12,11 @@ import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import java.io.PrintWriter;
 import java.sql.Connection;
 import java.sql.ConnectionBuilder;
+import java.sql.PreparedStatement;
 import java.sql.SQLException;
 import java.sql.SQLFeatureNotSupportedException;
-import java.sql.Statement;
-import java.util.logging.Logger;
 import javax.sql.DataSource;
+import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
@@ -30,7 +30,8 @@ import org.slf4j.LoggerFactory;
 public class MultiTenantDataSource implements DataSource {
 
     private final DataSource dataSource;
-    private static final org.slf4j.Logger log = LoggerFactory.getLogger(MultiTenantDataSource.class);
+    private static final Logger log = LoggerFactory.getLogger(MultiTenantDataSource.class);
+    private static final String SET_SEARCH_PATH_STATEMENT = "SET search_path TO ?";
 
     @SuppressFBWarnings("EI")
     public MultiTenantDataSource(DataSource dataSource) {
@@ -76,7 +77,7 @@ public class MultiTenantDataSource implements DataSource {
     }
 
     @Override
-    public Logger getParentLogger() throws SQLFeatureNotSupportedException {
+    public java.util.logging.Logger getParentLogger() throws SQLFeatureNotSupportedException {
         return dataSource.getParentLogger();
     }
 
@@ -106,12 +107,14 @@ public class MultiTenantDataSource implements DataSource {
      */
     @SuppressFBWarnings("SQL_INJECTION_JDBC")
     private static void setSearchPath(Connection connection) throws SQLException {
-        try (Statement statement = connection.createStatement()) {
-            String tenantSchemaDefinitionStatement = "SET search_path TO " + TenantContext.getCurrentDatabaseSchema();
+        String currentDatabaseSchema = TenantContext.getCurrentDatabaseSchema();
 
-            statement.execute(tenantSchemaDefinitionStatement);
+        try (PreparedStatement statement = connection.prepareStatement(SET_SEARCH_PATH_STATEMENT)) {
+            statement.setString(1, currentDatabaseSchema);
 
-            log.trace("Executing SQL: " + tenantSchemaDefinitionStatement);
+            statement.execute();
+
+            log.trace("Executed SQL: {} {}", SET_SEARCH_PATH_STATEMENT, currentDatabaseSchema);
         }
     }
 }

--- a/server/ee/libs/config/tenant-multi-data-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantDataSource.java
+++ b/server/ee/libs/config/tenant-multi-data-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantDataSource.java
@@ -7,17 +7,9 @@
 
 package com.bytechef.ee.tenant.multi.sql;
 
-import com.bytechef.tenant.TenantContext;
+import com.bytechef.tenant.sql.BaseDataSource;
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-import java.io.PrintWriter;
-import java.sql.Connection;
-import java.sql.ConnectionBuilder;
-import java.sql.PreparedStatement;
-import java.sql.SQLException;
-import java.sql.SQLFeatureNotSupportedException;
 import javax.sql.DataSource;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
  * Multi-tenant DataSource wrapper that sets the PostgreSQL search_path based on tenant context.
@@ -27,94 +19,11 @@ import org.slf4j.LoggerFactory;
  * @author Ivica Cardic
  * @author Igor Beslic
  */
-public class MultiTenantDataSource implements DataSource {
-
-    private final DataSource dataSource;
-    private static final Logger log = LoggerFactory.getLogger(MultiTenantDataSource.class);
-    private static final String SET_SEARCH_PATH_STATEMENT = "SET search_path TO ";
+public class MultiTenantDataSource extends BaseDataSource {
 
     @SuppressFBWarnings("EI")
     public MultiTenantDataSource(DataSource dataSource) {
-        this.dataSource = dataSource;
+        super(dataSource);
     }
 
-    @Override
-    public Connection getConnection() throws SQLException {
-        Connection connection = dataSource.getConnection();
-
-        setSearchPath(connection);
-
-        return connection;
-    }
-
-    @Override
-    public Connection getConnection(String username, String password) throws SQLException {
-        Connection connection = dataSource.getConnection(username, password);
-
-        setSearchPath(connection);
-
-        return connection;
-    }
-
-    @Override
-    public PrintWriter getLogWriter() throws SQLException {
-        return dataSource.getLogWriter();
-    }
-
-    @Override
-    public void setLogWriter(PrintWriter out) throws SQLException {
-        dataSource.setLogWriter(out);
-    }
-
-    @Override
-    public void setLoginTimeout(int seconds) throws SQLException {
-        dataSource.setLoginTimeout(seconds);
-    }
-
-    @Override
-    public int getLoginTimeout() throws SQLException {
-        return dataSource.getLoginTimeout();
-    }
-
-    @Override
-    public java.util.logging.Logger getParentLogger() throws SQLFeatureNotSupportedException {
-        return dataSource.getParentLogger();
-    }
-
-    @Override
-    public ConnectionBuilder createConnectionBuilder() throws SQLException {
-        return dataSource.createConnectionBuilder();
-    }
-
-    @Override
-    public <T> T unwrap(Class<T> iface) throws SQLException {
-        return dataSource.unwrap(iface);
-    }
-
-    @Override
-    public boolean isWrapperFor(Class<?> iface) throws SQLException {
-        return dataSource.isWrapperFor(iface);
-    }
-
-    /**
-     * Sets the PostgreSQL search_path to the current tenant's schema.
-     *
-     * <p>
-     * <b>Security Note:</b> The SQL injection suppression is for schema name injection which cannot be parameterized in
-     * PostgreSQL. Schema names must be included as identifiers, not as parameterized values. The tenant ID is validated
-     * to contain only alphanumeric characters, underscores, and hyphens to prevent SQL injection. This is a PostgreSQL
-     * limitation, not a security vulnerability.
-     */
-    @SuppressFBWarnings("SQL_INJECTION_JDBC")
-    private static void setSearchPath(Connection connection) throws SQLException {
-        String currentDatabaseSchema = TenantContext.getCurrentDatabaseSchema();
-
-        try (PreparedStatement statement =
-            connection.prepareStatement(SET_SEARCH_PATH_STATEMENT + currentDatabaseSchema)) {
-
-            statement.execute();
-
-            log.trace("Executed SQL: {} {}", SET_SEARCH_PATH_STATEMENT, currentDatabaseSchema);
-        }
-    }
 }

--- a/server/ee/libs/config/tenant-multi-pgvector-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantPgVectorDataSource.java
+++ b/server/ee/libs/config/tenant-multi-pgvector-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantPgVectorDataSource.java
@@ -19,7 +19,8 @@ import java.util.logging.Logger;
 import javax.sql.DataSource;
 
 /**
- * Multi-tenant DataSource wrapper for PgVector that sets the PostgreSQL search_path to the tenant's vectorstore schema.
+ * Multi-tenant DataSource wrapper for PgVector that sets the PostgreSQL search_path to the tenant's vector store
+ * schema.
  *
  * @version ee
  *
@@ -27,8 +28,8 @@ import javax.sql.DataSource;
  */
 public class MultiTenantPgVectorDataSource implements DataSource {
 
-    private static final String VECTORSTORE_SCHEMA_SUFFIX = "vectorstore";
-    private static final String SET_SEARCH_PATH_STATEMENT = "SET search_path TO ?";
+    private static final String VECTOR_STORE_SCHEMA_SUFFIX = "vectorstore";
+    private static final String SET_SEARCH_PATH_STATEMENT = "SET search_path TO ";
 
     private final DataSource dataSource;
 
@@ -106,10 +107,10 @@ public class MultiTenantPgVectorDataSource implements DataSource {
      */
     @SuppressFBWarnings("SQL_INJECTION_JDBC")
     private static void setSearchPath(Connection connection) throws SQLException {
-        String currentDatabaseSchema = TenantContext.getCurrentDatabaseSchema(VECTORSTORE_SCHEMA_SUFFIX);
+        String currentDatabaseSchema = TenantContext.getCurrentDatabaseSchema(VECTOR_STORE_SCHEMA_SUFFIX);
 
-        try (PreparedStatement statement = connection.prepareStatement(SET_SEARCH_PATH_STATEMENT)) {
-            statement.setString(1, currentDatabaseSchema);
+        try (PreparedStatement statement =
+            connection.prepareStatement(SET_SEARCH_PATH_STATEMENT + currentDatabaseSchema)) {
 
             statement.execute();
         }

--- a/server/ee/libs/config/tenant-multi-pgvector-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantPgVectorDataSource.java
+++ b/server/ee/libs/config/tenant-multi-pgvector-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantPgVectorDataSource.java
@@ -12,9 +12,9 @@ import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import java.io.PrintWriter;
 import java.sql.Connection;
 import java.sql.ConnectionBuilder;
+import java.sql.PreparedStatement;
 import java.sql.SQLException;
 import java.sql.SQLFeatureNotSupportedException;
-import java.sql.Statement;
 import java.util.logging.Logger;
 import javax.sql.DataSource;
 
@@ -28,6 +28,7 @@ import javax.sql.DataSource;
 public class MultiTenantPgVectorDataSource implements DataSource {
 
     private static final String VECTORSTORE_SCHEMA_SUFFIX = "vectorstore";
+    private static final String SET_SEARCH_PATH_STATEMENT = "SET search_path TO ?";
 
     private final DataSource dataSource;
 
@@ -105,9 +106,12 @@ public class MultiTenantPgVectorDataSource implements DataSource {
      */
     @SuppressFBWarnings("SQL_INJECTION_JDBC")
     private static void setSearchPath(Connection connection) throws SQLException {
-        try (Statement statement = connection.createStatement()) {
-            statement
-                .execute("SET search_path TO " + TenantContext.getCurrentDatabaseSchema(VECTORSTORE_SCHEMA_SUFFIX));
+        String currentDatabaseSchema = TenantContext.getCurrentDatabaseSchema(VECTORSTORE_SCHEMA_SUFFIX);
+
+        try (PreparedStatement statement = connection.prepareStatement(SET_SEARCH_PATH_STATEMENT)) {
+            statement.setString(1, currentDatabaseSchema);
+
+            statement.execute();
         }
     }
 }

--- a/server/ee/libs/config/tenant-multi-pgvector-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantPgVectorDataSource.java
+++ b/server/ee/libs/config/tenant-multi-pgvector-config/src/main/java/com/bytechef/ee/tenant/multi/sql/MultiTenantPgVectorDataSource.java
@@ -7,15 +7,8 @@
 
 package com.bytechef.ee.tenant.multi.sql;
 
-import com.bytechef.tenant.TenantContext;
+import com.bytechef.tenant.sql.BaseDataSource;
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-import java.io.PrintWriter;
-import java.sql.Connection;
-import java.sql.ConnectionBuilder;
-import java.sql.PreparedStatement;
-import java.sql.SQLException;
-import java.sql.SQLFeatureNotSupportedException;
-import java.util.logging.Logger;
 import javax.sql.DataSource;
 
 /**
@@ -25,94 +18,19 @@ import javax.sql.DataSource;
  * @version ee
  *
  * @author Ivica Cardic
+ * @author Igor Beslic
  */
-public class MultiTenantPgVectorDataSource implements DataSource {
+public class MultiTenantPgVectorDataSource extends BaseDataSource {
 
     private static final String VECTOR_STORE_SCHEMA_SUFFIX = "vectorstore";
-    private static final String SET_SEARCH_PATH_STATEMENT = "SET search_path TO ";
-
-    private final DataSource dataSource;
 
     @SuppressFBWarnings("EI")
     public MultiTenantPgVectorDataSource(DataSource dataSource) {
-        this.dataSource = dataSource;
+        super(dataSource);
     }
 
     @Override
-    public Connection getConnection() throws SQLException {
-        Connection connection = dataSource.getConnection();
-
-        setSearchPath(connection);
-
-        return connection;
-    }
-
-    @Override
-    public Connection getConnection(String username, String password) throws SQLException {
-        Connection connection = dataSource.getConnection(username, password);
-
-        setSearchPath(connection);
-
-        return connection;
-    }
-
-    @Override
-    public PrintWriter getLogWriter() throws SQLException {
-        return dataSource.getLogWriter();
-    }
-
-    @Override
-    public void setLogWriter(PrintWriter out) throws SQLException {
-        dataSource.setLogWriter(out);
-    }
-
-    @Override
-    public void setLoginTimeout(int seconds) throws SQLException {
-        dataSource.setLoginTimeout(seconds);
-    }
-
-    @Override
-    public int getLoginTimeout() throws SQLException {
-        return dataSource.getLoginTimeout();
-    }
-
-    @Override
-    public Logger getParentLogger() throws SQLFeatureNotSupportedException {
-        return dataSource.getParentLogger();
-    }
-
-    @Override
-    public ConnectionBuilder createConnectionBuilder() throws SQLException {
-        return dataSource.createConnectionBuilder();
-    }
-
-    @Override
-    public <T> T unwrap(Class<T> iface) throws SQLException {
-        return dataSource.unwrap(iface);
-    }
-
-    @Override
-    public boolean isWrapperFor(Class<?> iface) throws SQLException {
-        return dataSource.isWrapperFor(iface);
-    }
-
-    /**
-     * Sets the PostgreSQL search_path to the current tenant's vectorstore schema.
-     *
-     * <p>
-     * <b>Security Note:</b> The SQL injection suppression is for schema name injection which cannot be parameterized in
-     * PostgreSQL. Schema names must be included as identifiers, not as parameterized values. The tenant ID is validated
-     * to contain only alphanumeric characters, underscores, and hyphens to prevent SQL injection. This is a PostgreSQL
-     * limitation, not a security vulnerability.
-     */
-    @SuppressFBWarnings("SQL_INJECTION_JDBC")
-    private static void setSearchPath(Connection connection) throws SQLException {
-        String currentDatabaseSchema = TenantContext.getCurrentDatabaseSchema(VECTOR_STORE_SCHEMA_SUFFIX);
-
-        try (PreparedStatement statement =
-            connection.prepareStatement(SET_SEARCH_PATH_STATEMENT + currentDatabaseSchema)) {
-
-            statement.execute();
-        }
+    protected String getVectorSchemaSuffix() {
+        return VECTOR_STORE_SCHEMA_SUFFIX;
     }
 }

--- a/server/libs/config/app-config/src/main/java/com/bytechef/config/ApplicationProperties.java
+++ b/server/libs/config/app-config/src/main/java/com/bytechef/config/ApplicationProperties.java
@@ -1486,7 +1486,7 @@ public class ApplicationProperties {
             }
 
             /** Vector store provider */
-            private Provider provider = Provider.PGVECTOR;
+            private Provider provider;
 
             /** PgVector configuration */
             private PgVector pgVector = new PgVector();

--- a/server/libs/core/rest/rest-impl/src/main/java/com/bytechef/web/rest/error/GlobalResponseEntityExceptionHandler.java
+++ b/server/libs/core/rest/rest-impl/src/main/java/com/bytechef/web/rest/error/GlobalResponseEntityExceptionHandler.java
@@ -73,8 +73,6 @@ public class GlobalResponseEntityExceptionHandler extends AbstractResponseEntity
     @ExceptionHandler(Throwable.class)
     @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
     public ResponseEntity<ProblemDetail> handleAnyException(final Throwable throwable, final WebRequest request) {
-        logger.error(throwable.getMessage(), throwable);
-
         Exception exception = getCauseException((Exception) throwable);
 
         HttpStatus httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
@@ -82,6 +80,8 @@ public class GlobalResponseEntityExceptionHandler extends AbstractResponseEntity
         if (exception instanceof IllegalArgumentException) {
             return handleIllegalArgumentExceptionException((IllegalArgumentException) exception, request);
         }
+
+        logger.error(throwable.getMessage(), throwable);
 
         return ResponseEntity
             .of(createProblemDetail(exception, httpStatus, exception.getMessage(), null, null, request))

--- a/server/libs/core/rest/rest-impl/src/main/java/com/bytechef/web/rest/error/GlobalResponseEntityExceptionHandler.java
+++ b/server/libs/core/rest/rest-impl/src/main/java/com/bytechef/web/rest/error/GlobalResponseEntityExceptionHandler.java
@@ -43,7 +43,7 @@ import org.springframework.web.context.request.WebRequest;
 @Order(Ordered.HIGHEST_PRECEDENCE)
 public class GlobalResponseEntityExceptionHandler extends AbstractResponseEntityExceptionHandler {
 
-    private static final Logger logger = LoggerFactory.getLogger(GlobalResponseEntityExceptionHandler.class);
+    private static final Logger log = LoggerFactory.getLogger(GlobalResponseEntityExceptionHandler.class);
 
     @ResponseStatus(HttpStatus.BAD_REQUEST)
     @ExceptionHandler(AbstractException.class)
@@ -63,7 +63,7 @@ public class GlobalResponseEntityExceptionHandler extends AbstractResponseEntity
     public ResponseEntity<ProblemDetail> handleNoSuchElementException(
         final NoSuchElementException exception, final WebRequest request) {
 
-        logger.error(exception.getMessage(), exception);
+        log.error(exception.getMessage(), exception);
 
         return ResponseEntity
             .of(createProblemDetail(exception, HttpStatus.NOT_FOUND, exception.getMessage(), null, null, request))
@@ -75,16 +75,15 @@ public class GlobalResponseEntityExceptionHandler extends AbstractResponseEntity
     public ResponseEntity<ProblemDetail> handleAnyException(final Throwable throwable, final WebRequest request) {
         Exception exception = getCauseException((Exception) throwable);
 
-        HttpStatus httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
-
         if (exception instanceof IllegalArgumentException) {
             return handleIllegalArgumentExceptionException((IllegalArgumentException) exception, request);
         }
 
-        logger.error(throwable.getMessage(), throwable);
+        log.error(throwable.getMessage(), throwable);
 
         return ResponseEntity
-            .of(createProblemDetail(exception, httpStatus, exception.getMessage(), null, null, request))
+            .of(createProblemDetail(
+                exception, HttpStatus.INTERNAL_SERVER_ERROR, exception.getMessage(), null, null, request))
             .build();
     }
 
@@ -93,7 +92,7 @@ public class GlobalResponseEntityExceptionHandler extends AbstractResponseEntity
     public ResponseEntity<ProblemDetail> handleIllegalArgumentExceptionException(
         final IllegalArgumentException illegalArgumentException, final WebRequest request) {
 
-        logger.error(illegalArgumentException.getMessage(), illegalArgumentException);
+        log.error(illegalArgumentException.getMessage(), illegalArgumentException);
 
         Exception exception = getCauseException(illegalArgumentException);
 
@@ -106,7 +105,7 @@ public class GlobalResponseEntityExceptionHandler extends AbstractResponseEntity
     public ResponseEntity<ProblemDetail> handleConcurrencyFailureException(
         final ConcurrencyFailureException exception, final WebRequest request) {
 
-        logger.error(exception.getMessage(), exception);
+        log.error(exception.getMessage(), exception);
 
         return ResponseEntity
             .of(createProblemDetail(exception, HttpStatus.CONFLICT, "Concurrency Failure", null, null, request))

--- a/server/libs/core/tenant/tenant-api/src/main/java/com/bytechef/tenant/TenantContext.java
+++ b/server/libs/core/tenant/tenant-api/src/main/java/com/bytechef/tenant/TenantContext.java
@@ -26,11 +26,13 @@ import org.springframework.util.Assert;
 
 /**
  * @author Ivica Cardic
+ * @author Igor Beslic
  */
 public class TenantContext {
 
     public static final String CURRENT_TENANT_ID = "CURRENT_TENANT_ID";
     public static final String DEFAULT_TENANT_ID = "public";
+    private static final String TENANT_PATTERN = TenantConstants.TENANT_PREFIX + "_";
 
     private static final ThreadLocal<String> currentTenant = ThreadLocal.withInitial(() -> DEFAULT_TENANT_ID);
     private static final Logger log = LoggerFactory.getLogger(TenantContext.class);
@@ -52,15 +54,21 @@ public class TenantContext {
     }
 
     public static String getCurrentDatabaseSchema() {
-        return Objects.equals(getCurrentTenantId(), DEFAULT_TENANT_ID)
-            ? DEFAULT_TENANT_ID
-            : TenantConstants.TENANT_PREFIX + "_" + getCurrentTenantId();
+        return getCurrentDatabaseSchema(null);
     }
 
-    public static String getCurrentDatabaseSchema(String suffix) {
-        return Objects.equals(getCurrentTenantId(), DEFAULT_TENANT_ID)
-            ? DEFAULT_TENANT_ID
-            : TenantConstants.TENANT_PREFIX + "_" + suffix + "_" + getCurrentTenantId();
+    public static String getCurrentDatabaseSchema(String vectorSchemaSuffix) {
+        if (Objects.equals(getCurrentTenantId(), DEFAULT_TENANT_ID)) {
+            return DEFAULT_TENANT_ID;
+        }
+
+        String currentSuffix = "";
+
+        if (Objects.nonNull(vectorSchemaSuffix)) {
+            currentSuffix = vectorSchemaSuffix + "_";
+        }
+
+        return TENANT_PATTERN + currentSuffix + getCurrentTenantId();
     }
 
     public static String getCurrentTenantId() {

--- a/server/libs/core/tenant/tenant-api/src/main/java/com/bytechef/tenant/sql/BaseDataSource.java
+++ b/server/libs/core/tenant/tenant-api/src/main/java/com/bytechef/tenant/sql/BaseDataSource.java
@@ -1,0 +1,139 @@
+/*
+ * Copyright 2025 ByteChef
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.bytechef.tenant.sql;
+
+import com.bytechef.tenant.TenantContext;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.ConnectionBuilder;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import javax.sql.DataSource;
+
+/**
+ * @author Igor Beslic
+ */
+public abstract class BaseDataSource implements DataSource {
+
+    private final DataSource dataSource;
+    private static final String SET_SEARCH_PATH_STATEMENT = "SET search_path TO ";
+
+    public BaseDataSource(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    public boolean isWrapperFor(Class<?> iface) throws SQLException {
+        return dataSource.isWrapperFor(iface);
+    }
+
+    @Override
+    public <T> T unwrap(Class<T> iface) throws SQLException {
+        return dataSource.unwrap(iface);
+    }
+
+    @Override
+    public java.util.logging.Logger getParentLogger() throws SQLFeatureNotSupportedException {
+        return dataSource.getParentLogger();
+    }
+
+    protected DataSource getDataSource() {
+        return dataSource;
+    }
+
+    @Override
+    public int getLoginTimeout() throws SQLException {
+        return dataSource.getLoginTimeout();
+    }
+
+    @Override
+    public void setLoginTimeout(int seconds) throws SQLException {
+        dataSource.setLoginTimeout(seconds);
+    }
+
+    @Override
+    public PrintWriter getLogWriter() throws SQLException {
+        return dataSource.getLogWriter();
+    }
+
+    @Override
+    public void setLogWriter(PrintWriter out) throws SQLException {
+        dataSource.setLogWriter(out);
+    }
+
+    @Override
+    public ConnectionBuilder createConnectionBuilder() throws SQLException {
+        return dataSource.createConnectionBuilder();
+    }
+
+    @Override
+    public Connection getConnection() throws SQLException {
+        Connection connection = dataSource.getConnection();
+
+        setSearchPath(connection);
+
+        return connection;
+    }
+
+    @Override
+    public Connection getConnection(String username, String password) throws SQLException {
+        Connection connection = dataSource.getConnection(username, password);
+
+        setSearchPath(connection);
+
+        return connection;
+    }
+
+    /**
+     * Sets the search_path to the current tenant's schema. Method handles setting the search path for both vector and
+     * classic schema.
+     *
+     * <p>
+     * <b>Note:</b> This method is responsible for setting the database tenant in multitenant environments. If one
+     * relies on this API class and does not depend on the multitenant database, this method can be safely ignored.
+     * </p>
+     * <p>
+     * <b>Security Note:</b> The SQL injection suppression is for schema name configuration statement that cannot be
+     * parameterized with PreparedStatement in PostgreSQL. Schema names must be included as identifiers, not as
+     * parameterized values. The tenant ID is validated to contain only alphanumeric characters, underscores, and
+     * hyphens to prevent SQL injection. This is a PostgreSQL limitation, not a security vulnerability.
+     * </p>
+     */
+    @SuppressFBWarnings("SQL_INJECTION_JDBC")
+    protected void setSearchPath(Connection connection) throws SQLException {
+        String currentDatabaseSchema = TenantContext.getCurrentDatabaseSchema(getVectorSchemaSuffix());
+
+        try (PreparedStatement statement =
+            connection.prepareStatement(SET_SEARCH_PATH_STATEMENT + currentDatabaseSchema)) {
+
+            statement.execute();
+        }
+    }
+
+    /**
+     * Each tenant might have both vector and classic schema. Vector schema suffix designates the schema where vector
+     * data is stored.
+     *
+     * @return prefix for vector schema
+     */
+    protected String getVectorSchemaSuffix() {
+        return null;
+    }
+
+}

--- a/server/libs/core/tenant/tenant-api/src/main/java/com/bytechef/tenant/sql/BaseDataSource.java
+++ b/server/libs/core/tenant/tenant-api/src/main/java/com/bytechef/tenant/sql/BaseDataSource.java
@@ -34,6 +34,7 @@ public abstract class BaseDataSource implements DataSource {
     private final DataSource dataSource;
     private static final String SET_SEARCH_PATH_STATEMENT = "SET search_path TO ";
 
+    @SuppressFBWarnings("EI")
     public BaseDataSource(DataSource dataSource) {
         this.dataSource = dataSource;
     }

